presto-native-execution/presto_cpp/main/operators/tests/BroadcastTest.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/Uri.h>
+#include "folly/init/Init.h"
+#include "presto_cpp/external/json/json.hpp"
+#include "presto_cpp/main/operators/BroadcastExchangeSource.h"
+#include "presto_cpp/main/operators/BroadcastWrite.h"
+#include "presto_cpp/main/operators/LocalPersistentShuffle.h"
+#include "presto_cpp/main/operators/PartitionAndSerialize.h"
+#include "velox/connectors/hive/HivePartitionFunction.h"
+#include "velox/exec/Exchange.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/exec/tests/utils/QueryAssertions.h"
+
+using namespace facebook::velox;
+using namespace facebook::presto;
+using namespace facebook::presto::operators;
+
+namespace facebook::presto::operators::test {
+class BroadcastTest : public exec::test::OperatorTestBase {
+ public:
+ protected:
+  void SetUp() override {
+    exec::test::OperatorTestBase::SetUp();
+    filesystems::registerLocalFileSystem();
+    exec::Operator::registerOperator(
+        std::make_unique<BroadcastWriteTranslator>());
+    exec::ExchangeSource::registerFactory(
+        BroadcastExchangeSource::createExchangeSource);
+  }
+
+  static std::string makeTaskId(const std::string& prefix, int num) {
+    auto url = fmt::format("batch/{}-{}", prefix, num);
+    return url;
+  }
+
+  std::shared_ptr<exec::Task> makeTask(
+      const std::string& taskId,
+      core::PlanNodePtr planNode,
+      int destination) {
+    auto queryCtx = std::make_shared<core::QueryCtx>(executor_.get());
+    core::PlanFragment planFragment{planNode};
+    return exec::Task::create(
+        taskId, std::move(planFragment), destination, std::move(queryCtx));
+  }
+
+  auto addBroadcastWriteNode(const std::string& basePath) {
```

Comment:
Consider moving this function into presto-native-execution/presto_cpp/main/operators/tests/PlanBuilder.h